### DISCUSSION TITLE
Incorporating rewrite or redirect for a fallback page from getStaticProps/getStaticPaths

### DISCUSSION THREAD

**Discussion by agohil001** [vercel/next.js#16458] [Help]
The specific use case I am trying to solve is that I want to statically generate all "valid" dynamic paths based on entries of a specific type found on a headless CMS. So if a user tries navigating to an invalid path, they would get redirected to a default page defined on a different static path. Ideally, I would set fallback to false so the fallback check isn't necessary, but really it's just a nice-to-have.

I have managed to trigger a client-side redirect on a dynamic route page component by assessing through next router whether a page is on a fallback route, but this requires rendering the initial page first--and in my case, since the page component requires data from a CMS call, requires turning these "required" props into "optional" props and essentially adding fallback-specific render logic. With `getServerSideProps`, I was able to trigger a server-side redirect when we were unable to retrieve the required data for these props, but I'm wondering if there is an analogous approach using `getStaticProps` and `getStaticPaths`?

One approach I have been curious about is using the incremental adoption approach for rewrites defined within `next.config.js` as illustrated [here](https://nextjs.org/docs/api-reference/next.config.js/rewrites). When I tried this myself, it still rendered the fallback when fallback is set to true and 404 when set to false. I was wondering if anyone has gotten this approach to work for them?

**Comment by emindeniz99** [comment]
I searched this, and I can't find any solution or approach. Currently, Next.js doesn't provide custom fallback logic.
And, I need this feature for my planned project.

edit: I found this discussion https://github.com/vercel/next.js/discussions/14890
But, it is not implemented
